### ISSUE TITLE
the center seems not be updated?

### ISSUE THREAD

**Issue by kakashidan** [singlasahil14/resnet-exps#1]
In the center loss function, you define as follows:
```
def _center_loss_fn(self, embeddings, labels):
    embedding_size = embeddings.get_shape().as_list()[-1]
    centers = tf.get_variable(name='centers', shape=[self.hps.num_classes, embedding_size],
                              initializer=tf.random_normal_initializer(stddev=0.1), trainable=False)
    label_indices = tf.argmax(labels, 1)
    centers_batch = tf.nn.embedding_lookup(centers, label_indices)
    center_loss = tf.nn.l2_loss(embeddings - centers_batch)/tf.to_float(tf.shape(embeddings)[0])
    new_centers = centers_batch - embeddings
    labels_unique, row_indices, counts = tf.unique_with_counts(label_indices)

    centers_update = tf.unsorted_segment_sum(new_centers, row_indices, tf.shape(labels_unique)[0])/tf.to_float(counts)
    centers = tf.scatter_sub(centers, labels_unique, self.hps.ALPHA*centers_update)
    return center_loss
```
the `centers` is the op to update the center of mini batch. But, this op seems not be added into `train_ops` ? Then, how the centers be updated?
